docs/appkit/shared/smart-sessions.mdx

```diff
@@ -0,0 +1,269 @@
+                }
+            }
+        }])
+        ```
+</TabItem>
+</Tabs>
+
+
+#### Return value 
+<Tabs
+	groupId=""
+	activeOptions={["Return value","Return value Example"]}
+>
+<TabItem value="Return value">
+        ```tsx
+        type PrepareCallsReturnValue = [{
+            preparedCalls: {
+                type: string
+                data: any
+                chainId: `0x${string}`
+            }
+            signatureRequest: {  
+                hash: `0x${string}`
+            }
+            context: `0x${string}`
+        }]
+        ```
+</TabItem>
+<TabItem value="Return value Example">
+        ```tsx
+        [{
+            preparedCalls: {
+                type: 'user-operation-v07', type
+                data: { // ...userOp
+                sender: '0x...',
+                ...
+                },
+                chainId: '0x01'
+            },
+            signatureRequest: {  
+                hash: '0x...' // user op hash in our case
+            },
+            context: '...' // params.capabilities.permissions.context in our case
+        }]
+        ```
+</TabItem>
+</Tabs>
+
+    . App developers are expected to Sign the `signatureRequest.hash` returned from `wallet_prepareCalls` call using the dApp key (secp256k1 or secp256r1)
+    
+    . Dapps makes the `wallet_sendPreparedCalls` JSON rpc call to wallet service. The RPC accepts prepared calls from the response to a `wallet_prepareCalls` request along with a signature, and returns an EIP-5792 call bundle ID.
+
+## Currently supported Permission types
+
+#### ContractCallPermission
+    
+    ```jsx
+    export enum ParamOperator {
+      EQUAL = 'EQUAL',
+      GREATER_THAN = 'GREATER_THAN',
+      LESS_THAN = 'LESS_THAN'
+    }
+    
+    export enum Operation {
+      Call = 'Call',
+      DelegateCall = 'DelegateCall'
+    }
+    
+    export type ArgumentCondition = {
+      operator: ParamOperator
+      value: `0x${string}`
+    }
+    
+    export type FunctionPermission = {
+      functionName: string
+      args?: ArgumentCondition[]
+      valueLimit?: `0x${string}`
+      operation?: Operation
+    }
+    export type ContractCallPermission = {
+      type: 'contract-call'
+      data: {
+        address: `0x${string}`
+        abi: Record<string, unknown>[]
+        functions: FunctionPermission[]
+      }
+    }
+    ```
+    
+#### NativeTokenRecurringAllowancePermission
+    
+    ```jsx
+    export type NativeTokenRecurringAllowancePermission = {
+      type: 'native-token-recurring-allowance'
+      data: {
+        allowance: `0x${string}`
+        start: number
+        period: number
+      }
+    }
+    ```
+ 
+#### ERC20RecurringAllowancePermission
+    
+    ```jsx
+    type ERC20RecurringAllowancePermission = {
+      type: 'erc20-recurring-allowance'
+      data: {
+        token: `0x${string}`
+        allowance: `0x${string}`
+        start: number
+        period: number
+      }
+    }
```

Comment:
These are supported but we don't really have the UI for it yet I believe. Will need to check and come back to you 🙏